_posts/2019-01-08-data-driven-static-resume.adoc

```diff
@@ -0,0 +1,202 @@
+LiquiDoc is a nascent API for the AJYL docstack, designed for building rich-text documents in many editions and formats, all from the same source, by invoking upstream tools (such as Asciidoctor, Liquid, and Jekyll processing engines).
+
+The Codewriting project's LiquiDoc link:https://github.com/briandominick/codewriting/blob/master/_configs/build-global.yml[build configuration file (config)] for this site is pretty long, but essentially three relevant stages generate both the HTML and PDF versions of the resumé.
+Let's look at the first such stage.
+
+.From _configs/build-global.yml -- LiquiDoc config file resume parsing stage
+[source,yaml]
+----
+include::assets/includes/build-global.yml[tags="parse-resume-example-snippet"]
+----
+
+Here LiquiDoc is instructed to parse the `dominick.yml` data with the `resume.asciidoc` template.
+This generates `resume.adoc`, an AsciiDoc source file containing its own output instructions, saved to the ephemeral `_build/pages/` directory.
+
+Whenever the `resume.adoc` file is processed by Asciidoctor, header information (“frontmatter” in Jekyll parlance) designates the output settings and variables.
+First, it establishes settings and default variables, which will be used in the Jekyll site generation.
+
+.From _build/pages/resume.adoc -- Key Jekyll frontmatter settings and content variables
+[source,asciidoc]
+----
+include::pages/resume.adoc[tags="key-settings-jekyll"]
+----
+
+This built file's header also contains information for the PDF rendering.
+
+.From _build/pages/resume.adoc -- Key PDF settings and content variables
+[source,asciidoc]
+----
+\ifdef::backend-pdf[]
+include::pages/resume.adoc[tags="key-settings-pdf"]
+\endif::backend-pdf[]
+----
+
+The first and last lines are conditionals; they ensure the enveloped settings are only applied during rendering procedures if the `backend-pdf` setting (“attribute” in AsciiDoc parlance) is defined, but they will replace any previously set parameters with the same names.
+
+Which gets us back to the LiquiDoc config.
+The next stage instructs LiquiDoc to actually render the PDF version of the output from the source file prebuilt in the last stage (`_build/pages/resume.adoc`).
+
+.From _configs/build-global.yml -- LiquiDoc config file resume PDF render stage
+[source,yaml]
+----
+include::assets/includes/build-global.yml[tags="render-resume-example-snippet"]
+----
+
+This block invokes a very simple Asciidoctor conversion using the link:https://github.com/asciidoctor/asciidoctor-pdf[PDF backend extension], which engages Prawn for the final conversion.
+A customized link:https://github.com/briandominick/codewriting/blob/master/theme/pdf/resume-theme.yml[YAML-formatted theme file] enforces the style of the PDF output.
+
+The resulting PDF artifact will later be copied into the Jekyll-generated website, but first we have to perform that Jekyll generation.
+The LiquiDoc config has us covered here, as well.
```

Comment:
This sounds interesting but as I reader I'm not quite sure what you mean.

Reply:
Rewriting this a bit.